adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -389,40 +385,13 @@ private[rdd] class AlignmentRecordRDDFunctions(val rdd: RDD[AlignmentRecord])
       // however! this function is not implemented on all platforms, hence the try.
       try {
 
-        // concat files together
-        // we need to do this through reflection because the concat method is
-        // NOT in hadoop 1.x
-
-        // find the concat method
-        val fsMethods = classOf[FileSystem].getDeclaredMethods
-        val filteredMethods = fsMethods.filter(_.getName == "concat")
-
-        // did we find the method? if not, throw an exception
-        if (filteredMethods.size == 0) {
-          throw new IllegalStateException(
-            "Could not find concat method in FileSystem. Methods included:\n" +
-              fsMethods.map(_.getName).mkString("\n") +
-              "\nAre you running Hadoop 1.x?")
-        } else if (filteredMethods.size > 1) {
-          throw new IllegalStateException(
-            "Found multiple concat methods in FileSystem:\n%s".format(
-              filteredMethods.map(_.getName).mkString("\n")))
-        }
-
-        // since we've done our checking, let's get the method now
-        val concatMethod = filteredMethods.head
-
         // we need to move the head file into the tailFiles directory
         // this is a requirement of the concat method
         val newHeadPath = new Path("%s/header".format(tailPath))
         fs.rename(headPath, newHeadPath)
 
-        // invoke the method on the fs instance
-        // if we are on hadoop 2.x, this makes the call:
-        //
-        // fs.concat(headPath, tailFiles)
         try {
-          concatMethod.invoke(fs, newHeadPath, tailFiles)
+          fs.concat(newHeadPath, tailFiles)
         } catch {
           case ite: InvocationTargetException => {
```

Comment:
This needs to change to `case t: Throwable => {`

A line further down will then need to change from `throw ite.getTargetException` to `throw t`